tsp/src/definitions/mod.rs

```diff
@@ -15,7 +13,7 @@ pub type NonConfidentialData<'a> = &'a [u8];
 pub type TSPMessage = Vec<u8>;
 
 #[cfg(feature = "async")]
-pub type TSPStream<E> = std::pin::Pin<Box<dyn Stream<Item = Result<BytesMut, E>> + Send>>;
+pub type TSPStream<D, E> = std::pin::Pin<Box<dyn Stream<Item = Result<D, E>> + Send>>;
```

Comment:
I'd suggest adding `dyn` in the type name somewhere, to clarify that it's a trait object